### DISCUSSION TITLE
New groups added to uwgroup file

### DISCUSSION THREAD

**Discussion by Richardvdg30** [trix7777/m2strm#15] [Q&A]
Is it possible to automaticly ad newly added groups from the m3u file to the uwgroupfile or automaticly ignore these goups from processing?
As it stands now when i procces the m3u file new groups are automaticly downloaded to the strm dir.
So now I first have to monitor the strm folder and remove the newly added groups before i can let Kodi update the library.

I hope you understand my question.

**Comment by trix7777** [comment]
It is most certainty doable. Could be a configuration option.
The main reason for making the app this way is to not miss out on any new groups without user intervention.

**Comment by Richardvdg30** [comment]
Thanks for your answer.
Obviously there alwas has to be some user intervention ;)

A config option would be great.
In that case just a quick check on the newgroups.log would mean running the exe again for the new group i would wantafter all  or no action at all.

9 out of 10 times the new groups are the ones I have no use for.
My m3u file has 250 groups from all over the world from which i only have 6 uncommented.

**Comment by trix7777** [comment]
Meanwhile I suggest you check this out:
[https://github.com/bebo-dot-dev/m3u-epg-editor](https://github.com/bebo-dot-dev/m3u-epg-editor)
I've been using it for a few years for filtering out TV-channels I don't want in my tvheadend backend, also just keeping the groups that I have specified.

**Comment by Richardvdg30** [comment]
I have been trying to get your suggestion to work but no succes.
Must be due to my lack of pyton knowhow.
I could not get the cfg_json file to work, and just ended up with a load of error from the version p3 file.

Your install instruction and cfg file are more straight forward and simple.
I'll wait until you come up with a sollution for m2strm.

**Comment by trix7777** [comment]
example command line, from shell or cron:
`/usr/bin/python3 /home/trix7777/m3u-epg-editor/m3u-epg-editor-py3.py --json_cfg /home/trix7777/m3u-epg-editor/args_tv.json`

example contents of args_tv.json with comments (comments are not really supported in python, but this works):
```
{
	"m3uurl": "http://leet.tv:1337/get.php?username=trix7777&password=1234g&type=m3u_plus&output=ts",
	"epgurl": "http://leet.tv:1337/xmltv.php?username=trix7777&password=1234",
	"groups": [
        	"^Sweden$",
        	"^United States$",
		"^United Kingdom$",
	],
	"groupmode": "keep",
	"discard_channels": [
		"^COMMENT-FG1$",
		"^COMMENT-DISCARD_ALL_CHANNELS_WITH_THE_FOLLOWING_KEYWORDS_IN_THEM$",
		"^COMMENT-UPSIGN_MEANS_MUSTSTARTWITH_AND_DOLLARSIGN_MUSTENDWITH$",
		"^COMMENT-DOT_AND_STAR_MEANS_ANY_CHAR_OR_NONE$",
        	"\\[Live During Events( Only|)\\]",
		"^-=.*=-",
		"^===.*===",
		
		"^COMMENT-FG2$",
		"^COMMENT-DISCARD_THE_FOLLOWING_SWEDISH_CHANNELS$",
		"^ATG (Live|Swedish Horse Racing)( HD| FHD| UHD|) SE$",
		"^World Poker Tour( HD| FHD| UHD|) SE$",
		
		"^COMMENT-FG3$",
		"^COMMENT-DISCARD_THE_FOLLOWING_US_CHANNELS$",
		"^(MAV|Mav)( |)TV( HD| FHD| UHD|) US$",
		"^Willow Cricket( HD| FHD| UHD|) US$",
		
		"^COMMENT-FG4$",
		"^COMMENT-DISCARD_THE_FOLLOWING_UK_CHANNELS$",
		".* \\+1( HD| FHD| UHD|) UK$",
		"^(E|eir)( Sport|Sport) [1-2]( HD| FHD| UHD|) UK$",
		"^Amazon UK Europsort.*( HD| FHD| UHD|) UK$"
		],
	"no_tvg_id": true,
	"no_epg": false,
	"no_sort": true,
    	"http_for_images": true,
	"preserve_case": true,
    	"outdirectory": "/home/trix7777/m3u-epg-editor",
    	"outfilename": "_tv_raw"
}
```

**Comment by Richardvdg30** [comment]
Thanks for your examples, I tried but could not get it to work.
Keep getting errors for the difrent dependencies on win10 and win11 pc's

`C:\Users\Richard\Desktop\M3U Editor>m3u-epg-editor-py3.py --json_cfg args_tv.json
Traceback (most recent call last):
  File "C:\Users\Richard\Desktop\M3U Editor\m3u-epg-editor-py3.py", line 27, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'

C:\Users\Richard\Desktop\M3U Editor>pip show requests
Name: requests
Version: 2.31.0
Summary: Python HTTP for Humans.
Home-page: https://requests.readthedocs.io
Author: Kenneth Reitz
Author-email: me@kennethreitz.org
License: Apache 2.0
Location: c:\users\richard\appdata\local\programs\python\python310\lib\site-packages
Requires: certifi, charset-normalizer, idna, urllib3
Required-by:`
 
I give up, been on it for hours....

**Comment by trix7777** [comment]
Working on a solution to this right now! See Ideas not yet implemented #18 for more information.
